algoliasearch/src/main/java/com/algolia/search/clients/AlgoliaConfig.java

```diff
@@ -0,0 +1,121 @@
+package com.algolia.search.clients;
+
+import com.algolia.search.transport.StatefulHost;
+import java.util.*;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.ForkJoinPool;
+
+/** * Algolia's clients configuration */
+@SuppressWarnings("WeakerAccess")
+public class AlgoliaConfig {
+
+  public AlgoliaConfig(String applicationID, String apiKey) {
+    this(applicationID, apiKey, ForkJoinPool.commonPool());
+  }
+
+  public AlgoliaConfig(String applicationID, String apiKey, ExecutorService executor) {
+    this.applicationID = applicationID;
+    this.apiKey = apiKey;
+    this.executor = executor;
+    this.batchSize = 1000;
+    this.defaultHeaders =
+        new HashMap<String, String>() {
+          {
+            put("X-Algolia-Application-Id", getApplicationID());
+            put("X-Algolia-API-Key", getApiKey());
+            put(
+                "User-Agent",
+                String.format("Algolia for Java (%s); JVM (%s)", clientVersion, javaVersion));
+            put("Accept", "application/json");
+          }
+        };
+  }
+
+  public String getApplicationID() {
+    return applicationID;
+  }
+
+  public AlgoliaConfig setApplicationID(String applicationID) {
```

Comment:
😢, mutability

Reply:
Good catch! I shouldn't be too confident with intelij generators 😅